client/lib/wp/offline-library/index.js

```diff
@@ -0,0 +1,102 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import stringify from 'json-stable-stringify';
+
+const readCache = () => {
+	try {
+		return require( '../../../../cached-requests.json' );
+	} catch ( e ) {
+		return {};
+	}
+};
+
+const saveRequests = requests => {
+	const requestData = {};
+
+	requests.forEach( ( value, key ) => {
+		const path = JSON.parse( key ).path;
+
+		if ( ! requestData.hasOwnProperty( path ) ) {
+			requestData[ path ] = {};
+		}
+
+		requestData[ path ][ key ] = value;
+	} );
+
+	const file = new Blob( [ JSON.stringify( requestData, null, 2 ) ], {
+		type: 'application/json',
+	} );
+
+	const url = URL.createObjectURL( file );
+	window.open( url );
+};
+
+export const makeOffline = wpcom => {
+	const location = window.location;
```

Comment:
nit, but what do you think about: 
```js
const queryParams = new URLSearchParams( locations.search );
const primingRequested = queryParams.has( 'wpcom_priming' );
```

Reply:
that's cool. no IE support but probably worth it. in this case I really don't care about any spec as it seems quite reasonable to just search for that literal string.

think anyone developing in IE will mind? maybe I'll update it.

Reply:
How about using feature flags for this?  If you do `config.isEnabled( 'offline/wpcom-priming' ) then `?flags=offline/wpcom-priming` will work [for free](https://github.com/Automattic/wp-calypso/blob/0ca94bf/client/config/index.js#L22-L38).

If that's not desirable for some reason, I would just keep the regex, and maybe make it more general to not require the `=1` part (`/[&?]wpcom_priming(=1|=true|&|$)/` for example).

Reply:
socked IE in 9107268